repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,263 @@
+# SPDX-FileCopyrightText: 2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+
+import base64
+import os
+import sys
+from hashlib import sha256
+from pathlib import Path
+from typing import Optional
+from urllib import request
+from urllib.parse import urlparse
+
+from click import Context
+from tuf.api.exceptions import DownloadError, RepositoryError
+from tuf.ngclient import Updater
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+from repository_service_tuf.helpers.hash_bins import find_hash_bin
+
+
+def update_artifact_name_with_hash_prefix(artifact_name: str) -> str:
+    artifact_file_name = artifact_name.split("/")[-1]
+    hash_prefix = find_hash_bin(artifact_file_name)
+    new_artifact_file_name = hash_prefix + "-" + artifact_file_name
+    splitted = artifact_name.rsplit("/", 1)
+    if len(splitted) > 1:
+        return artifact_name.rsplit("/", 1)[0] + "/" + new_artifact_file_name
+    else:
+        # Annotating as it's covered, but codecov doesn't
+        # recognize that properly
+        return new_artifact_file_name  # pragma: no cover
+
+
+def decode_trusted_root(root) -> str:
```

Comment:
Do we finally make it `Any`?

Reply:
base64 library cannot use a parameter, encoded as `any`